planner/core/logical_plan_builder.go

```diff
@@ -3599,7 +3608,7 @@ func (b *PlanBuilder) buildDataSource(ctx context.Context, tn *ast.TableName, as
 	if tblName.L == "" {
 		tblName = tn.Name
 	}
-	possiblePaths, err := getPossibleAccessPaths(b.ctx, b.TableHints(), tn.IndexHints, tbl, dbName, tblName)
+	possiblePaths, err := getPossibleAccessPaths(b.ctx, b.TableHints(), tn.IndexHints, tbl, dbName, tblName, b.isForUpdateRead, b.is.SchemaMetaVersion())
```

Comment:
`buildTableRefsWithCache`'s cache mechanism should be take care...

for sql: 

```
create table t1(id int);
create table t2(id int);
select (select id from t2 limit 1 for update), id from t1;
```

the value in this line should be:

|  tbl   | b.isForUpdateRead  |
|  ----  | ----  |
| t1  | false |
| t2  | true |

but current got:

|  tbl   | b.isForUpdateRead  |
|  ----  | ----  |
| t1  | false |
| t2  | false |

Reply:
Good catch, added some cases for `resolveSelect`.